composer.json

```
@@ -22,14 +22,14 @@
 		}
 	},
 	"require" : {
-		"php" : "^7.3",
-		"symfony/options-resolver": "^3.4|^4.1|^5.0",
-		"symfony/yaml": "^3.4|^4.1|^5.0"
+		"php" : ">=7.3",
```

This must not include the future PHP 9, otherwise when PHP 9 will finally be released this very old version of lib will be installable instead of the proper future version of this lib.

(that sounds a bit overkill, but this kind of problem actually do happen. Currently it exist in PHP-Cs-Fixer where a very old version was declared compatible with PHP 8, which is obviously incorrect and lead to crash)

```suggestion
		"php" : "^7.3 || ^8.0",
```

---

Thanks for your input @PowerKiKi! I understand your concern. I would personally also use your approach (in fact I did in my first version of this PR), but I also respect the view the Symfony community has about this. In their approach, they do not want to restrict testing of libraries for future php versions, even though they primarily would not support it (in a php version 9 or 10) and I guess every develop who would include this very package in his project has the freedom to set a version constraint which restricts the usage to a php version compatible release. In the end, it is up to the maintainer to decide. Maybe @jschaedl has some input on that as well, but as he requested the change to it, I think he personally prefers the Symfony approach.

---

> restrict testing

Very good point. And I have found myself annoyed by such restrictions in the past. But I think it should be workaround-able via `--ignore-platform-reqs` flag.

Let see what @jschaedl thinks of that...

---

Don't know if you know about this, but there is also a very interesting interview with Nikolas Grekas (Symfony core team) about this on YT: https://www.youtube.com/watch?v=c3bpTBjhK2Y

---

I have to say I didn't notice that it was explicitly asked to change this way. So most likely my suggestion can be ignored.

Thanks for the link, I'll take a look !